server/src/main/java/org/opensearch/cluster/metadata/Metadata.java

```diff
@@ -260,7 +281,7 @@ public interface Custom extends NamedDiffable<Custom>, ToXContentFragment, Clust
     private final Settings settings;
     private final DiffableStringMap hashesOfConsistentSettings;
     private final Map<String, IndexMetadata> indices;
-    private final Map<String, IndexTemplateMetadata> templates;
+    private final TemplatesMetadata templates;
```

Comment:
Wouldnt changing the existing cluster state structure break backward compatibility ?

Reply:
The newly created TemplateMetadata has exactly the same parsing. So, this new class will still be backward compatible with older map while serializing and de-serializing.

Reply:
Should we include an IT to verify this?

Reply:
Multiple Tests in ClusterStateTests already exists which verify the serialized value has not changed even after this change. Will explore if we can add a BwC test for this as well.

Reply:
yes, a bwc test would help us verify that nothing breaks

Reply:
why do we need the wrapper class anyway?

Reply:
All the metadata attribute files we are creating are implementing ToXContentFragment interface, to maintain uniformity for all these metadata attribute.

Reply:
I have added a bwc test as well, to ensure that when version upgrade happens, old nodes are able to join the cluster